openedx/features/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -0,0 +1,964 @@
+            self.userC,
+            [self.orgA.short_name, self.orgB.short_name],
+        )
+
+        self.st1 = Taxonomy.objects.create(
+            name="st1", system_defined=True, enabled=True
+        )
+        TaxonomyOrg.objects.create(
+            taxonomy=self.st1,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+            org=None,
+        )
+        self.st2 = Taxonomy.objects.create(
+            name="st2", system_defined=True, enabled=False
+        )
+        TaxonomyOrg.objects.create(
+            taxonomy=self.st2,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+
+        self.t1 = Taxonomy.objects.create(name="t1", enabled=True)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.t1,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        self.t2 = Taxonomy.objects.create(name="t2", enabled=False)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.t2,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        # Orphaned taxonomy
+        # self.t3 = Taxonomy.objects.create(name="t3", enabled=True)
+        # self.t4 = Taxonomy.objects.create(name="t4", enabled=False)
+
+        self.tA1 = Taxonomy.objects.create(name="tA1", enabled=True)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tA1,
+            org=self.orgA,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        self.tA2 = Taxonomy.objects.create(name="tA2", enabled=False)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tA2,
+            org=self.orgA,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+
+        self.tB1 = Taxonomy.objects.create(name="tB1", enabled=True)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tB1,
+            org=self.orgB,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        self.tB2 = Taxonomy.objects.create(name="tB2", enabled=False)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tB2,
+            org=self.orgB,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+
+        self.tC1 = Taxonomy.objects.create(name="tC1", enabled=True)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tC1,
+            org=self.orgA,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tC1,
+            org=self.orgB,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        self.tC2 = Taxonomy.objects.create(name="tC2", enabled=False)
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tC2,
+            org=self.orgA,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+        TaxonomyOrg.objects.create(
+            taxonomy=self.tC2,
+            org=self.orgB,
+            rel_type=TaxonomyOrg.RelType.OWNER,
+        )
+
+
+@ddt.ddt
+@override_settings(FEATURES={"ENABLE_CREATOR_GROUP": True})
+class TestTaxonomyViewSet(TestTaxonomyViewSetMixin, APITestCase):
```

Comment:
This isn't your fault @rpenido , but this change really highlights how writing tests to match the acceptance criteria (as I suggested you do) makes the tests themselves illegible.

I've raised https://github.com/openedx/modular-learning/issues/89 as a separate issue to refactor these tests (and `test_rules.py`) to remove most of the `ddt.data` confusion, and make the tests more legible, as @bradenmacdonald  has suggested.

Reply:
Copying what i wrote below:

@pomegranited Just my 2 cents here: When my brain was fried trying to figure out which permissions were wrong, I found it easier to deal with a large dtt function than a helper function.

The helper function will stop in the first assert failure. Sometimes I fixed the first condition only to discover that the second condition also failed and got trapped, changing the behavior without a clear view of all failing conditions.

The ddt always showed me all mismatched permissions.

Maybe I was just more comfortable with ddt because I wrote it (I'm not sure).

I just want to give you this insight. I'm following your lead here and don't disagree with your approach.

There's a discussion about this somewhere (besides the issue you created)? I'm interested in the results and want to follow.